server/src/test/java/io/crate/auth/HttpAuthUpstreamHandlerTest.java

```diff
@@ -346,4 +351,89 @@ public void test_user_authentication_with_jwt_token_verified_per_connection() th
         verify(jwtAuth, times(1)).authenticate(any(Credentials.class), any(ConnectionProperties.class));
     }
 
+    @Test
+    public void test_user_has_no_jwt_properties_default_is_used() throws Exception {
```

Comment:
I'd really prefer a unit test which just tests the relevant `JWTAuthenticationMethod.authenticate` method. I think this would also avoid all the mocks.

Reply:
In https://github.com/crate/crate/commit/12622b548b796184c71ed32ad698996b1a42bbae we have different lookups, and it's decided before JWTAuthentication kicks in

Reply:
>We already have such test in (UserAuthenticationMethodTest.test_jwt_authentication_user_has_no_jwt_properties_default_is_used)

I actually can remove it and keep this one - as it cover auth and also correct usage of Credentials/lookup

Reply:
Hm so this test is effectively testing one line, the `globalJwtEnabled ? credentials.tokenMatch() : credentials.jwtPropertyMatch();`. This is not so great. With my suggestion on moving this decision into the `Credentials.matchByToken()`, we can remove this test and use a dedicated unit test.